beaker/ext/redisclusternm.py

```diff
@@ -0,0 +1,158 @@
+import os
+import threading
+import time
+import pickle
+
+try:
+    import redis
+except ImportError:
+    redis = None
+
+from beaker.container import NamespaceManager
+from beaker.synchronization import SynchronizerImpl
+from beaker.util import SyncDict, machine_identifier
+from beaker.crypto.util import sha1
+from beaker._compat import string_type, PY2
+
+
+class RedisClusterNamespaceManager(NamespaceManager):
+    """Provides the :class:`.NamespaceManager` API over Redis cluster.
+
+    Provided ``url`` can be both a redis connection string or
+    an already existing RedisCluster instance.
+
+    The data will be stored into redis keys, with their name
+    starting with ``beaker_cache:``. So make sure you provide
+    a specific database number if you don't want to mix them
+    with your own data.
+    """
+    MAX_KEY_LENGTH = 1024
+
+    clients = SyncDict()
+
+    def __init__(self, namespace, url, timeout=None, **kw):
```

Comment:
- Rename url to urls
- add a docstring explaining that urls must be comma separated + gives an example, something like `urls: set list of redis cluster urls separated by a comma. Example: https://192.168.1.2:6679,https://192.168.1.3:6679,https://192.168.1.4:6679`
- add explanation about not putting a db name in the url

Reply:
fixed.